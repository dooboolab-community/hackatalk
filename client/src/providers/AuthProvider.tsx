import React, {useReducer} from 'react';

import {User} from '../types/graphql';
import createCtx from '../utils/createCtx';

export interface State {
  user: User | undefined;
}

interface Context {
  state: State;
  setUser(user: User | undefined): void;
}

const [useCtx, Provider] = createCtx<Context>();

// eslint-disable-next-line no-shadow
export enum ActionType {
  SetUser = 'set-user',
}

type SetUserAction = {
  type: ActionType.SetUser;
  payload: {user: User};
};
type Action = SetUserAction;

interface Props {
  children?: React.ReactElement;
  initialAuthUser?: User;
}

type Reducer = (state: State, action: Action) => State;

const setUser = (dispatch: React.Dispatch<SetUserAction>) => (
  authUser: User,
): void => {
  dispatch({
    type: ActionType.SetUser,
    payload: {user: authUser},
  });
};

const initialState: State = {
  user: undefined,
};

// eslint-disable-next-line default-param-last
const reducer: Reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SetUser:
      return {...action.payload};
    default:
      return state;
  }
};

function AuthProvider(props: Props): React.ReactElement {
  const initialAuthState = {
    user: props.initialAuthUser,
  };

  const [state, dispatch] = useReducer<Reducer>(reducer, initialAuthState);

  const actions = {
    setUser: setUser(dispatch),
  };

  return <Provider value={{state, ...actions}}>{props.children}</Provider>;
}

const AuthContext = {
  useAuthContext: useCtx,
  AuthProvider,
};

export {useCtx as useAuthContext, AuthProvider};
export default AuthContext;
