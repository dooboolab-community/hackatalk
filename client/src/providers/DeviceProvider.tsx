import * as Device from 'expo-device';

import React, {FC, useEffect, useState} from 'react';

import createCtx from '../utils/createCtx';

interface Context {
  deviceType: Device.DeviceType;
  setDeviceType: React.Dispatch<React.SetStateAction<Device.DeviceType>>;
}

const [useCtx, Provider] = createCtx<Context>();

const DeviceProvider: FC = ({children}) => {
  const [deviceType, setDeviceType] = useState<Device.DeviceType>(
    Device.DeviceType.PHONE,
  );

  useEffect(() => {
    let isMounted = true;

    Device.getDeviceTypeAsync().then((result) => {
      // Only change state if this component is still mounted.
      if (isMounted) setDeviceType(result);
    });

    return () => {
      isMounted = false;
    };
  }, []);

  return (
    <Provider
      value={{
        deviceType,
        setDeviceType,
      }}>
      {children}
    </Provider>
  );
};

const DeviceContext = {
  useDeviceContext: useCtx,
  DeviceProvider,
};

export {useCtx as useDeviceContext, DeviceProvider};
export default DeviceContext;
