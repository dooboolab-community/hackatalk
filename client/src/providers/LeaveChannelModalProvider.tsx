import React, {createContext, useContext, useState} from 'react';
import type {ReactElement, ReactNode} from 'react';

type LeaveChannelModalData = {
  testID?: string;
  channelId: string;
  onClickYes?: () => void;
};

export type LeaveChannelModalState =
  | {isVisible: false}
  | ({isVisible: true} & LeaveChannelModalData);

export type LeaveChannelModalContext = {
  showModal: (next: LeaveChannelModalData) => void;
  hideModal: () => void;
  modalState: LeaveChannelModalState;
};

const LeaveChannelModalContext = createContext<LeaveChannelModalContext>({
  modalState: {isVisible: false},
  showModal: () => {},
  hideModal: () => {},
});

type Props = {
  children: ReactNode;
};

export const LeaveChannelModalProvider = ({children}: Props): ReactElement => {
  const [showModalState, setModalState] = useState<LeaveChannelModalState>({
    isVisible: false,
  });

  const showModal = (next: LeaveChannelModalData): void =>
    setModalState({
      isVisible: true,
      ...next,
    });

  const hideModal = (): void =>
    setModalState({
      isVisible: false,
    });

  return (
    <LeaveChannelModalContext.Provider
      value={{
        showModal,
        hideModal,
        modalState: showModalState,
      }}
    >
      {children}
    </LeaveChannelModalContext.Provider>
  );
};

export const useLeaveChannelContext = (): LeaveChannelModalContext =>
  useContext(LeaveChannelModalContext);
