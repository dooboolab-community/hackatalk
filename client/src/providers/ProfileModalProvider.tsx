import React, {FC, createContext, useContext, useState} from 'react';

import {ProfileModal_user$key} from '../__generated__/ProfileModal_user.graphql';

export type ModalState =
  | {
      isVisible: false;
    }
  | {
      /** Modal visiblity */
      isVisible: true;
      /** Which user the profile modal describes */
      user: ProfileModal_user$key;
      /** Is the profile user a friend of the current user? */
      isFriend?: boolean;
      /** Callback function for delete button */
      onDeleteFriend?: () => void;
      /** Callback function for add friend button */
      onAddFriend?: () => void;
      hideButtons?: boolean;
    };

export type ProfileModalContext = {
  /** Allow context consumers to make the modal visible with a new state. */
  showModal: (next: Omit<ModalState & {isVisible: true}, 'isVisible'>) => void;
  /** Allow context consumers to hide the modal. */
  hideModal: () => void;
  modalState: ModalState;
};

const ProfileModalContext = createContext<ProfileModalContext>({
  modalState: {isVisible: false},
  showModal: () => {},
  hideModal: () => {},
});

export const ProfileModalProvider: FC = (props) => {
  const [state, setState] = useState<ModalState>({
    isVisible: false,
  });

  const showModal = (
    next: Omit<ModalState & {isVisible: true}, 'isVisible'>,
  ): void =>
    setState({
      isVisible: true,
      ...next,
    });

  const hideModal = (): void =>
    setState({
      isVisible: false,
    });

  return (
    <ProfileModalContext.Provider
      value={{
        showModal,
        hideModal,
        modalState: state,
      }}>
      {props.children}
    </ProfileModalContext.Provider>
  );
};

export const useProfileContext = (): ProfileModalContext =>
  useContext(ProfileModalContext);
