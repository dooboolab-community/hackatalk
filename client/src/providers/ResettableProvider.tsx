import React, {FC, useState} from 'react';

import {IEnvironment} from 'relay-runtime';
import {RelayEnvironmentProvider} from 'react-relay/hooks';
import createCtx from '../utils/createCtx';

interface Context {
  /**
   * Current Relay environment.
   */
  environment: IEnvironment;

  /**
   * Discard the current Relay environment and
   * create a new one.
   */
  resetRelayEnvironment: () => void;
}

interface Props {
  /* For testing */
  environment?: IEnvironment;

  /* Factory function for generating Relay environments */
  createRelayEnvironment: () => IEnvironment;
}

const [useResettableRelayContext, Provider] = createCtx<Context>();

const ResettableRelayProvider: FC<Props> = ({
  children,
  environment: initialEnvironment,
  createRelayEnvironment,
}) => {
  const [environment, setEnvironment] = useState<IEnvironment>(
    initialEnvironment || createRelayEnvironment(),
  );

  const resetRelayEnvironment = (): void => {
    setEnvironment(createRelayEnvironment());
  };

  return (
    <Provider value={{environment, resetRelayEnvironment}}>
      <RelayEnvironmentProvider environment={environment}>
        {children}
      </RelayEnvironmentProvider>
    </Provider>
  );
};

export {useResettableRelayContext, ResettableRelayProvider};
