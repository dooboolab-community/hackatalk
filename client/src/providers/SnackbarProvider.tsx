import React, {
  ReactElement,
  ReactNode,
  createContext,
  useCallback,
  useContext,
  useRef,
  useState,
} from 'react';
import {Snackbar, SnackbarContent, SnackbarRef} from 'dooboo-ui';

import {Theme} from '../theme';
import {View} from 'react-native';

export interface SnackbarState extends SnackbarContent {
  testID?: string;
  zIndex?: number;
  theme?: Theme;
}

export type SnackbarContext = {
  openSnackbar: (newSnackbarState: SnackbarState) => void;
};

const SnackbarContext = createContext<SnackbarContext>({
  openSnackbar: () => {},
});

type Props = {
  children: ReactNode;
};

export const SnackbarProvider = (props: Props): ReactElement => {
  const [snackbarState, setSnackbarState] = useState<SnackbarState>();

  const snackbarRef = useRef<SnackbarRef>(null);

  const openSnackbar = useCallback(
    (newSnackbarState: SnackbarState): void => {
      setSnackbarState(newSnackbarState);

      const {content, timer, styles, actionText, onActionPress, type} =
        newSnackbarState;

      snackbarRef.current?.show({
        content,
        timer,
        styles,
        actionText,
        onActionPress,
        type,
      });
    },
    [snackbarRef, setSnackbarState],
  );

  return (
    <SnackbarContext.Provider
      value={{
        openSnackbar,
      }}
    >
      {props.children}
      <View style={{zIndex: snackbarState?.zIndex}}>
        <Snackbar testID={snackbarState?.testID} ref={snackbarRef} />
      </View>
    </SnackbarContext.Provider>
  );
};

export const useSnackbarContext = (): SnackbarContext =>
  useContext(SnackbarContext);
