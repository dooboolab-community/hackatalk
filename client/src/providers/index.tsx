export * from './AuthProvider';
export * from './DeviceProvider';
export * from './ProfileModalProvider';
