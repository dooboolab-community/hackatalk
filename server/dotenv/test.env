JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
