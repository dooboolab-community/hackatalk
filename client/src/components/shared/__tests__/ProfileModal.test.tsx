import React, {
  ForwardRefExoticComponent,
  RefAttributes,
  createRef,
  forwardRef,
  useImperativeHandle,
} from 'react';
import {
  RenderAPI,
  act,
  fireEvent,
  render,
  waitFor,
} from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import Shared from '../ProfileModal';
import { useProfileContext } from '../../../providers/ProfileModalProvider';

type Handle<T> = T extends ForwardRefExoticComponent<RefAttributes<infer T2>>
  ? T2
  : never;
interface Ref {
  showModal: (showModalParams: any) => void;
}
interface Props {
  onAddFriend?: () => void;
  onDeleteFriend?: () => void;
}

// interface Props {}
const FakeProfileModal = forwardRef<Ref, Props>((props, ref) => {
  const { state, showModal } = useProfileContext();
  const modalEl = React.useRef(null);

  state.modal = modalEl;

  useImperativeHandle(ref, () => ({
    showModal,
    // modalEl,
  }));

  return (
    <Shared
      ref={modalEl}
      onAddFriend={props.onAddFriend}
      onDeleteFriend={props.onDeleteFriend}
    />
  );
});

describe('[ProfileModal] rendering test', () => {
  let props: any;
  let component: React.ReactElement;
  let testingLib: RenderAPI;
  let fakeProfileModalRef;
  let testingLib2;
  let component2;

  const ref = createRef<Handle<typeof Shared>>();
  const onAddFriend = jest.fn();
  const onDeleteFriend = jest.fn();

  beforeEach(() => {
    props = createTestProps();

    component = createTestElement(
      <Shared ref={ref} {...props} />,
    );

    testingLib = render(component);
    fakeProfileModalRef = createRef<Ref>();

    component2 = createTestElement(
      <FakeProfileModal
        ref={fakeProfileModalRef}
        onAddFriend={onAddFriend}
        onDeleteFriend={onDeleteFriend}
      />,
    );

    testingLib2 = render(component2);
  });

  it('Render without crashing', async () => {
    testingLib = render(component);

    await waitFor(() => {
      const json = testingLib.toJSON();

      expect(json).toMatchSnapshot();
    });
  });

  it('Should be opened', async () => {
    const { current } = ref;

    // Open modal by using it's state.
    await act(async () => {
      current?.open();
    });

    const button = testingLib.queryByTestId('touch-add-friend');

    expect(button).not.toBeNull();
  });

  it('Check "Added to your friend." button', async () => {
    const { current } = ref;

    await act(async () => {
      current?.open();
      current?.setIsFriendAdded(true);
    });

    const button = testingLib.queryByTestId('added-message');

    expect(button).not.toBeNull();
  });

  it('Should be closed', async () => {
    const { current } = ref;

    // Cloase modal by using it's state.
    await act(async () => {
      current?.close();
    });

    const button = testingLib.queryByTestId('touch-add-friend');

    expect(button).toBeNull();
  });

  it('delete', async () => {
    const { current } = ref;

    await act(async () => {
      current?.showAddBtn(false);
      current?.open();
    });

    const button = testingLib.queryByTestId('touch-add-friend');

    await act(async () => {
      fireEvent.press(button);
    });

    expect(current?.modal).toBeTruthy();
  });

  // it('Add Friend', async () => {
  //   await act(async () => {
  //     fakeProfileModalRef.current?.showModal({
  //       user: { photoURL: '', nickname: 'nickname', statusMessage: 'online' },
  //     });
  //   });

  //   const button = testingLib2.queryByTestId('touch-add-friend');

  //   await act(async () => {
  //     fireEvent.press(button);
  //   });

  //   expect(onAddFriend).toHaveBeenCalled();
  // });

  // it('Delete Friend', async () => {
  //   await act(async () => {
  //     fakeProfileModalRef.current?.showModal({
  //       user: { photoURL: '', nickname: 'nickname', statusMessage: 'online' },
  //       deleteMode: true,
  //     });
  //   });

  //   const button = testingLib2.queryByTestId('touch-add-friend');

  //   await act(async () => {
  //     fireEvent.press(button);
  //   });

  //   expect(onDeleteFriend).toHaveBeenCalled();
  // });
});
