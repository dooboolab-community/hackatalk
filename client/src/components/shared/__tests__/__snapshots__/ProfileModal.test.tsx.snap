// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ProfileModal] rendering test Render without crashing 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View />
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": "100%",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingBottom": 4,
          "paddingEnd": 4,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <TouchableWithoutFeedback
        hitSlop={
          Object {
            "bottom": 4,
            "left": 4,
            "right": 4,
            "top": 4,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(250, 186, 48, 0.95)",
              },
              Object {
                "borderRadius": 14,
                "height": 28,
                "justifyContent": "center",
                "paddingHorizontal": 12,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 12,
                "includeFontPadding": false,
                "lineHeight": 16,
              }
            }
          >
            Dismiss All
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
    <ScrollView
      data={
        Array [
          Object {
            "category": "Animated.event now requires a second argument for options",
            "warnings": Array [
              YellowBoxWarning {
                "message": Object {
                  "content": "Animated.event now requires a second argument for options",
                  "substitutions": Array [],
                },
                "stack": Array [
                  Object {
                    "arguments": Array [],
                    "column": 29,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/YellowBox/Data/YellowBoxWarning.js",
                    "lineNumber": 58,
                    "methodName": "Function.parse",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 56,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/YellowBox/YellowBox.js",
                    "lineNumber": 156,
                    "methodName": "registerWarning",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 23,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/YellowBox/YellowBox.js",
                    "lineNumber": 88,
                    "methodName": "warnImpl",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 12,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/YellowBox/YellowBox.js",
                    "lineNumber": 66,
                    "methodName": "BufferedConsole.warn",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 14,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/Animated/src/AnimatedEvent.js",
                    "lineNumber": 98,
                    "methodName": "new AnimatedEvent",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 24,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/Animated/src/AnimatedImplementation.js",
                    "lineNumber": 526,
                    "methodName": "Object.event",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 29,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native-modalbox/index.js",
                    "lineNumber": 366,
                    "methodName": "ModalBox.createPanResponder",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 16,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native-modalbox/index.js",
                    "lineNumber": 118,
                    "methodName": "new ModalBox",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 17,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-test-renderer/cjs/react-test-renderer.development.js",
                    "lineNumber": 3629,
                    "methodName": "constructClassInstance",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 4,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-test-renderer/cjs/react-test-renderer.development.js",
                    "lineNumber": 7558,
                    "methodName": "updateClassComponent",
                  },
                ],
                "symbolicated": Object {
                  "error": null,
                  "stack": null,
                  "status": "NONE",
                },
              },
              YellowBoxWarning {
                "message": Object {
                  "content": "Animated.event now requires a second argument for options",
                  "substitutions": Array [],
                },
                "stack": Array [
                  Object {
                    "arguments": Array [],
                    "column": 29,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/YellowBox/Data/YellowBoxWarning.js",
                    "lineNumber": 58,
                    "methodName": "Function.parse",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 56,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/YellowBox/YellowBox.js",
                    "lineNumber": 156,
                    "methodName": "registerWarning",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 23,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/YellowBox/YellowBox.js",
                    "lineNumber": 88,
                    "methodName": "warnImpl",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 12,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/YellowBox/YellowBox.js",
                    "lineNumber": 66,
                    "methodName": "BufferedConsole.warn",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 14,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/Animated/src/AnimatedEvent.js",
                    "lineNumber": 98,
                    "methodName": "new AnimatedEvent",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 24,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native/Libraries/Animated/src/AnimatedImplementation.js",
                    "lineNumber": 526,
                    "methodName": "Object.event",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 29,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native-modalbox/index.js",
                    "lineNumber": 366,
                    "methodName": "ModalBox.createPanResponder",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 16,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-native-modalbox/index.js",
                    "lineNumber": 118,
                    "methodName": "new ModalBox",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 17,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-test-renderer/cjs/react-test-renderer.development.js",
                    "lineNumber": 3629,
                    "methodName": "constructClassInstance",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 4,
                    "file": "/Users/dooboolab/Github/dooboolab/hackatalk/client/node_modules/react-test-renderer/cjs/react-test-renderer.development.js",
                    "lineNumber": 7558,
                    "methodName": "updateClassComponent",
                  },
                ],
                "symbolicated": Object {
                  "error": null,
                  "stack": null,
                  "status": "NONE",
                },
              },
            ],
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      maxToRenderPerBatch={10}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      scrollEnabled={false}
      scrollEventThrottle={50}
      scrollsToTop={false}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "height": 48.5,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View
        style={null}
      >
        <TouchableWithoutFeedback>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(250, 186, 48, 0.95)",
                },
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "marginTop": 0.5,
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.5)",
                    "fontSize": 14,
                    "includeFontPadding": false,
                    "lineHeight": 18,
                  }
                }
              >
                (2) 
              </Text>
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 1)",
                    "flex": 1,
                    "fontSize": 14,
                    "includeFontPadding": false,
                    "lineHeight": 18,
                  }
                }
              >
                <Text>
                  Animated.event now requires a second argument for options
                </Text>
              </Text>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </ScrollView>
    <SafeAreaView
      style={
        Object {
          "backgroundColor": "rgba(250, 186, 48, 0.95)",
          "marginTop": 0.5,
        }
      }
    />
  </View>
</View>
`;
